### PR TITLE
Update Nix files for LLDB Pwndbg

### PR DESCRIPTION
**Pull request by mbrla0**
This PR updates our Nix files for LLDB Pwndbg support, under a new package, named `pwndbg-lldb`. It also updates `pwndbg-lldb.py` to handle different Pwndbg locations. We pick version 19 of LLDB in our flake, because when running under earlier versions we [hit an LLDB bug specific to Python >=3.12](https://github.com/llvm/llvm-project/issues/70453), which is the version of Python in the flake.

Feedback on my Nix is also especially appreciated, as I'm by no means a Nix guru.

### CODE REVIEW

File: nix/pwndbg.nix
Comment (patryk4815) [inline nix/pwndbg.nix:59]:
```suggestion
    nativeBuildInputs = [ pkgs.makeWrapper ];
    buildInputs = [ pyEnv ];
```

File: nix/pwndbg.nix
Comment (patryk4815) [inline nix/pwndbg.nix]:
```suggestion
```
Not needed when we have `buildInputs`

File: nix/pwndbg.nix
Comment (patryk4815) [inline nix/pwndbg.nix]:
```suggestion
      wrapProgram $out/bin/${pwndbgName} \
        --prefix PATH : ${ pkgs.lib.makeBinPath [ lldb ] } \
        '' + (pkgs.lib.optionalString !pkgs.stdenv.isDarwin ''
        --set LLDB_DEBUGSERVER_PATH ${ pkgs.lib.makeBinPath [ lldb ] }/lldb-server \
        '') + ''
        --set PWNDBG_LLDBINIT_DIR $out/share/pwndbg
```
- for darwin `LLDB_DEBUGSERVER_PATH` env must be unset
- for linux would be nice to use `set-default` it allows overriding 

it looks little ugly

**Review by patryk4815 — CHANGES_REQUESTED** [review]

File: nix/pwndbg.nix
Comment (mbrla0) [reply nix/pwndbg.nix]: Yeah, it does look ugly for sure. Also, we probably don't want to let Linux users override it, since, in general, we probably want the debug server to always match the LLDB package we use. Darwin is an exception, because we have no other choice. Overriding the variable should probably be done by customizing the Nix files or overriding the derivation.

Comment (patryk4815) [reply nix/pwndbg.nix]: Ok 👍 Edited proposal

File: nix/pwndbg.nix
Comment (mbrla0) [reply nix/pwndbg.nix]: Removed it.

**Comment by disconnect3d** [comment]
Waiting for @patryk4815 approval

**Review by patryk4815 — APPROVED** [review]
lgtm :)

**Pull request merged**
